delfin/drivers/utils/ssh_client.py

```diff
@@ -0,0 +1,147 @@
+#    WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+#    License for the specific language governing permissions and limitations
+#    under the License.
+import os
+
+import paramiko as paramiko
+
+from oslo_log import log as logging
+
+from delfin import exception
+
+LOG = logging.getLogger(__name__)
+
+
+class SSHClient(object):
+    """Common class for Hpe 3parStor storage system."""
+    SOCKET_TIMEOUT = 30
+
+    def __init__(self, **kwargs):
+
+        ssh_access = kwargs.get('ssh')
+        if ssh_access is None:
+            raise exception.InvalidInput('Input ssh_access is missing')
+        self.ssh_host = ssh_access.get('host')
+        self.ssh_port = ssh_access.get('port')
+        self.ssh_username = ssh_access.get('username')
+        self.ssh_password = ssh_access.get('password')
+        self.ssh_private_key = ssh_access.get('host_key')
+        self.ssh_conn_timeout = ssh_access.get('conn_timeout')
+        if self.ssh_conn_timeout is None:
+            self.ssh_conn_timeout = SSHClient.SOCKET_TIMEOUT
+
+    def connect(self):
+        self.ssh = paramiko.SSHClient()
+
+        if self.ssh_private_key is None:
+            self.ssh.set_missing_host_key_policy(paramiko.AutoAddPolicy())
+        else:
+            path = os.getcwd()
+            filename = path + '\\' + self.ssh_host + 'known_hosts'
+            file = open(filename, 'w')
+            file.write(self.ssh_private_key)
+            file.close()
+            self.ssh.load_host_keys(filename)
+
+        self.ssh.connect(hostname=self.ssh_host, port=self.ssh_port,
+                         username=self.ssh_username,
+                         password=self.ssh_password,
+                         timeout=self.ssh_conn_timeout)
+
+    def execCommand(self, command_str):
+        result = None
+        try:
+            if command_str is not None:
+                if self.ssh is not None:
+                    stdin, stdout, stderr = self.ssh.exec_command(command_str)
+                    res, err = stdout.read(), stderr.read()
+                    re = res if res else err
+                    result = re.decode()
+        except Exception as e:
+            LOG.error(e)
+            result = e
+        return result
+
+    def close(self):
+        try:
+            if self.ssh is not None:
+                # Close connection
+                self.ssh.close()
+                self.ssh = None
+        except Exception as e:
+            LOG.error(e)
+
+    def doexec(self, context, command_str):
+        """Clear alert from storage system."""
+        re = None
+        try:
+            if command_str is not None:
+                self.connect()
+                re = self.execCommand(command_str)
+        except paramiko.AuthenticationException as ae:
+            LOG.error('doexec Authentication error:{}'.format(ae))
+            raise exception.InvalidUsernameOrPassword()
```

Comment:
Please rename exception from InvalidUsernameOrPassword to AuthenticationException